applications/StatisticsApplication/custom_utilities/method_utilities.cpp

```diff
@@ -0,0 +1,775 @@
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(rData.size() != rReferenceData.size())
+        << "Data size and reference data vector size mismatch. [ "
+        << rData.size() << " != " << rReferenceData.size() << " ].\n";
+
+    KRATOS_CATCH("");
+}
+
+template <>
+void DataTypeSizeChecker(const Matrix& rData, const Matrix& rReferenceData)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(rData.size1() != rReferenceData.size1())
+        << "Data size and reference data matrix size1 mismatch. [ "
+        << rData.size1() << " != " << rReferenceData.size1() << " ].\n";
+    KRATOS_ERROR_IF(rData.size2() != rReferenceData.size2())
+        << "Data size and reference data matrix size2 mismatch. [ "
+        << rData.size2() << " != " << rReferenceData.size2() << " ].\n";
+
+    KRATOS_CATCH("");
+}
+
+template <>
+NodesContainerType& GetLocalDataContainer(ModelPart& rModelPart)
+{
+    return rModelPart.GetCommunicator().LocalMesh().Nodes();
+}
+
+template <>
+ElementsContainerType& GetLocalDataContainer(ModelPart& rModelPart)
+{
+    return rModelPart.GetCommunicator().LocalMesh().Elements();
+}
+
+template <>
+ConditionsContainerType& GetLocalDataContainer(ModelPart& rModelPart)
+{
+    return rModelPart.GetCommunicator().LocalMesh().Conditions();
+}
+
+template <>
+const NodesContainerType& GetLocalDataContainer(const ModelPart& rModelPart)
+{
+    return rModelPart.GetCommunicator().LocalMesh().Nodes();
+}
+
+template <>
+const ElementsContainerType& GetLocalDataContainer(const ModelPart& rModelPart)
+{
+    return rModelPart.GetCommunicator().LocalMesh().Elements();
+}
+
+template <>
+const ConditionsContainerType& GetLocalDataContainer(const ModelPart& rModelPart)
+{
+    return rModelPart.GetCommunicator().LocalMesh().Conditions();
+}
+
+template <>
+NodesContainerType& GetDataContainer(ModelPart& rModelPart)
+{
+    return rModelPart.Nodes();
+}
+
+template <>
+ElementsContainerType& GetDataContainer(ModelPart& rModelPart)
+{
+    return rModelPart.Elements();
+}
+
+template <>
+ConditionsContainerType& GetDataContainer(ModelPart& rModelPart)
+{
+    return rModelPart.Conditions();
+}
+
+template <>
+const NodesContainerType& GetDataContainer(const ModelPart& rModelPart)
+{
+    return rModelPart.Nodes();
+}
+
+template <>
+const ElementsContainerType& GetDataContainer(const ModelPart& rModelPart)
+{
+    return rModelPart.Elements();
+}
+
+template <>
+const ConditionsContainerType& GetDataContainer(const ModelPart& rModelPart)
+{
+    return rModelPart.Conditions();
+}
+
+double GetDoubleValue(const std::string& rInput)
+{
+    KRATOS_TRY
+    const int string_length = rInput.size();
+
+    KRATOS_ERROR_IF(string_length == 0)
+        << "Empty string provided, where double value is required.\n";
+
+    const int digit_length =
+        std::count_if(rInput.begin(), rInput.end(),
+                      [](unsigned char c) { return std::isdigit(c); });
+    const int seperator_length = std::count_if(
```

Comment:
I used sth like this and it did not compile with an old version of gcc ( I think 4.8.4) Probably not a problem here